### PR TITLE
Added `getkey` method to return original case-sensitive key.

### PR DESCRIPTION
**Pull request by justinjeffery-ipf**
Poetry Update.
Updated Docs.
Added tests.

```py
>>> from typing import Union

>>> from case_insensitive_dict import CaseInsensitiveDict

>>> case_insensitive_dict = CaseInsensitiveDict[Union[str, int], str](data={"Aa": "b", 1: "c"})
>>> case_insensitive_dict.getkey("aa")
'Aa'
```

### CODE REVIEW

**Comment by justinjeffery-ipf** [comment]
Need to update version.  Bump to `0.3.0` ?

**Review by rikhilrai — APPROVED** [review]

**Comment by rikhilrai** [comment]
> Need to update version. Bump to `0.3.0` ?

yeah, looks good :)

**Comment by rikhilrai** [comment]
going to merge the other open PRs from @dependabot and merge this on top

File: case_insensitive_dict/case_insensitive_dict.py [hunk 1]
```diff
@@ -50,14 +50,17 @@ def _convert_key(key: KT) -> KT:
...
+
     def __setitem__(self, key: KT, value: VT) -> None:
         self._data[self._convert_key(key=key)] = (key, value)
 
     def __getitem__(self, key: KT) -> VT:
-        try:
-            return self._data[self._convert_key(key=key)][1]
-        except KeyError:
-            raise KeyError(f"Key: {key!r} not found.") from None
+        return self._get_key_value(key)[1]
```
Comment (rikhilrai) [inline case_insensitive_dict/case_insensitive_dict.py:63]:
```suggestion
        return self._get_key_value(key=key)[1]
```

File: case_insensitive_dict/case_insensitive_dict.py [hunk 2]
```diff
@@ -80,6 +83,9 @@ def __eq__(self, other: Any) -> bool:
     def copy(self) -> CaseInsensitiveDict[KT, VT]:
         return CaseInsensitiveDict(data=dict(self._data.values()))
 
+    def getkey(self, key: KT) -> KT:
+        return self._get_key_value(key)[0]
```
Comment (rikhilrai) [inline case_insensitive_dict/case_insensitive_dict.py:87]:
```suggestion
        return self._get_key_value(key=key)[0]
```

**Comment by rikhilrai** [comment]
@justinjeffery-ipf, a version bump as well please

**Comment by justinjeffery-ipf** [comment]
I have no write access to main so if you want to merge you will also have to do so.

Thanks!

**Pull request merged**